File: BmiCalculation.java

import java.util.Scanner;

public class BmiCalculation {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double height;
        double weight;
        if (args.length >= 2) {
            height = Double.parseDouble(args[0]);
            weight = Double.parseDouble(args[1]);
            if (height <= 0 || weight <= 0) throw new IllegalArgumentException("Height and weight must be positive");
        } else {
            height = readPositiveDouble(scanner);
            weight = readPositiveDouble(scanner);
        }

        double bmi = weight / (height * height);
        System.out.printf("%.2f%n", bmi);
        scanner.close();
    }

    private static double readPositiveDouble(Scanner scanner) {
        while (true) {
            if (!scanner.hasNext()) throw new IllegalStateException("No more input");
            String token = scanner.next();
            try {
                double v = Double.parseDouble(token);
                if (v > 0) return v;
                // continue prompting silently by looping
            } catch (NumberFormatException e) {
                // ignore and continue reading
            }
        }
    }
}

File: BMI Calculation.py
import sys

height= int(input("Enter the Height (m) : "))
weight= float(input("Enter the weight (Kg) : "))

BMI = weight / (height * height ) 
print(f"Your BMI is : {BMI:.2f}")

if BMI <= 18.5:
    print("Your BMI is UnderWeight :")

elif BMI <= 24.5:
    print("Your BMI is Normal :")
elif BMI <= 29.9:
    print("Your BMI is Overweight :")
else:
    print("Your BMI is Obesity  :")